File: examples/sbp_dr.jl
using Dimred, Statistics

include("read.jl")

# Reduce to complete cases
v = [
    :BPXSY1,
    :RIAGENDR,
    :RIDAGEYR,
    :BMXWT,
    :BMXHT,
    :BMXBMI,
    :BMXLEG,
    :BMXARML,
    :BMXARMC,
    :BMXWAIST,
    :BMXHIP,
]
dx = df[:, v]
dx = dx[completecases(dx), :]
disallowmissing!(dx)

# Recode the sex variable to numeric
dx[!, :RIAGENDRx] = Float64.(replace(dx[:, :RIAGENDR], "F" => 1, "M" => -1))
v = replace(v, :RIAGENDR => :RIAGENDRx)
dx = dx[:, v]

# Center all variables
for m in names(dx)
    if eltype(dx[:, m]) <: Real
        dx[!, m] = dx[:, m] .- mean(dx[:, m])
    end
end

# Get design matrix and response vector
xv = [u for u in v if u != :BPXSY1]
xx = Matrix{Float64}(dx[:, xv])
yy = Vector{Float64}(dx[:, :BPXSY1])

# Sort the cases according to the response variable
ii = sortperm(yy)
yy = yy[ii]
xx = xx[ii, :]

# Fit a model using sliced inverse regression
ms = fit(SlicedInverseRegression, xx, yy)

# Use chi^2 tests for the dimension
pvs = dimension_test(ms)

# Fit a model using principal Hessian directions
mp = fit(PrincipalHessianDirections, xx, yy)

# Use chi^2 tests for the dimension
pvp = dimension_test(mp)

# Fit a model using sliced average variance estimation
ma = fit(SlicedAverageVarianceEstimation, xx, yy; ndir=5)

# Use chi^2 tests for the dimension
pva = dimension_test(ma)

File: examples/sbp_dr.R
library(dplyr)
library(ggplot2)
library(readr)
#library(dr)
source("../test/dr/R/dr.R")
source("read.R")

# Reduce to complete cases
dx = df %>% dplyr::select(BPXSY1, RIAGENDR, RIDAGEYR, BMXWT, BMXHT, BMXBMI, BMXLEG, BMXARML, BMXARMC, BMXWAIST, BMXHIP)
dx = dx[complete.cases(dx),]

# Recode sex to numeric
dx$RIAGENDRx = recode(dx$RIAGENDR, "F"=1, "M"=-1)

# Center all variables
for (m in names(dx)) {
    if (is.numeric(dx[[m]])) {
        dx[[m]] = scale(dx[[m]], scale=F)
    }
}

# Fit a model using sliced inverse regression
ms = dr(BPXSY1 ~ RIAGENDRx + RIDAGEYR + BMXWT + BMXHT + BMXBMI + BMXLEG + BMXARML + BMXARMC + BMXWAIST + BMXHIP, dx)

# Use chi^2 tests for the dimension
print(dr.test(ms))

# Fit a model using principal Hessian directions
mp = dr(BPXSY1 ~ RIAGENDRx + RIDAGEYR + BMXWT + BMXHT + BMXBMI + BMXLEG + BMXARML + BMXARMC + BMXWAIST + BMXHIP, dx, method="phd")

# Use chi^2 tests for the dimension
print(dr.test(mp))

# Fit a model using sliced average variance estimation
ma = dr(BPXSY1 ~ RIAGENDRx + RIDAGEYR + BMXWT + BMXHT + BMXBMI + BMXLEG + BMXARML + BMXARMC + BMXWAIST + BMXHIP, dx, method="save")

# Use chi^2 tests for the dimension
print(dr.test(ma))
